### PR TITLE
Task Editor: Part 1

### PR DESCRIPTION
**Pull request by jabbawookiees**

### CODE REVIEW

**Comment by jabbawookiees** [comment]
Merged this because other people might need some parts in a hurry. None of it works for now anyway.

File: src/client/components/mathjax.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// Just use this same config everywhere please
```
Comment (TimDumol) [inline src/client/components/mathjax.tsx:1]: You'll only see this comment if you happen to find this XD In an ideal world, (1) prolly wrap `MathJaxContext` in a component that uses this fixed `MathJaxConfig` and then (2) have a linter that bans direct usage of `MathJaxContext`; step (2) being significantly harder than (1)

File: src/client/components/task_editor/task_editor.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    </MathJaxContext>
+  );
+};
+
+type TaskTitleDisplayProps = {
+  title: string;
+  slug: string;
+};
+
+const TaskTitleDisplay = memo(({ title, slug }: TaskTitleDisplayProps) => {
```
Comment (TimDumol) [inline src/client/components/task_editor/task_editor.tsx:69]: AFAIK this `memo` call is useless since React already re-renders only on prop changes

**Review by TimDumol — COMMENTED** [review]
I randomly felt like judging your code heh

File: src/client/components/task_editor/task_editor.tsx (hunk 2 above)
Comment (jabbawookiees) [reply src/client/components/task_editor/task_editor.tsx:69]: I'm fairly certain that's not true. React will re-render the whole tree down if you don't use memo anywhere (or use `React.PureComponent`)

Citation: https://www.joshwcomeau.com/react/why-react-re-renders/#its-not-about-the-props-2

File: src/client/components/mathjax.tsx (hunk 1 above)
Comment (jabbawookiees) [reply src/client/components/mathjax.tsx:1]: MathJax is jank anyway! CJ is making me use KaTeX (which I didn't know about before he made a comment!)

File: src/client/components/task_editor/task_editor.tsx (hunk 2 above)
Comment (TimDumol) [reply src/client/components/task_editor/task_editor.tsx:69]: I stand corrected 😮

File: src/client/components/mathjax.tsx (hunk 1 above)
Comment (TimDumol) [reply src/client/components/mathjax.tsx:1]: back in my day we used `jsMath`! you youngin's and your new toys

**Pull request merged**